specs/AppLifecycle/Activation/AppLifecycle Activation.md

```diff
@@ -0,0 +1,248 @@
+
+    // Register for toast activation,
+    // where g_ToastGuid is defined as a static guid.
+    registrationManager.RegisterForToastActivation("Contoso Toast"); 
+}
+```
+### Get rich activation objects
+A Win32 app would typically get its command-line arguments very early in its WinMain, and we therefore expect the app to call the new AppLifecycle::GetActivatedEventArgs in the same place, instead of using the supplied lpCmdLine parameter or using GetCommandLineW. From there, the app can determine the specific ActivationKind, query the object for the specific XXXActivatedEventArgs type, and then access the properties and methods of that type.
+
+```c++
+void RespondToActivation()
+{
+    IActivatedEventArgs args = AppLifecycle::GetActivatedEventArgs();
+
+    if (args)
+    {
+        ActivationKind kind = args.Kind();
+        if (kind == ActivationKind::Launch)
+        {
+            auto launchArgs = args.as<LaunchActivatedEventArgs>();
+            DoSomethingWithLaunchArgs(launchArgs.Arguments());
+        }
+        else if (kind == ActivationKind::File)
+        {
+            auto fileArgs = args.as<FileActivatedEventArgs>();
+            DoSomethingWithFileArgs(fileArgs.Files());
+        }
+        else if (kind == ActivationKind::Protocol)
+        {
+            auto protocolArgs = args.as<ProtocolActivatedEventArgs>();
+            DoSomethingWithProtocolArgs(protocolArgs.Uri());
+        }
+        else if (kind == ActivationKind::StartupTask)
+        {
+            auto startupArgs = args.as<StartupTaskActivatedEventArgs>();
+            DoSomethingWithStartupArgs(startupArgs.TaskId());
+        }
+        else if (kind == ActivationKind::ToastNotification)
+        {
+            auto toastArgs = args.as<ToastNotificationActivatedEventArgs>();
+            DoSomethingWithToastArgs(toastArgs.Argument());
+        }
+    }
+}
+```
+### Unregister
+The primary requirement is to be able to register for the supported activation kinds dynamically. A secondary requirement is to be able to unregister for any or all of these.
+
+For filetype associations, the app can register for multiple filetypes. We will provide the ability to unregister one or more filetypes. For protocol registrations, the app can register for multiple protocols individually, so the API will also enable the app to unregister for each of these individually. For each of the other supported activation kinds, there can be at most one registration for the app. Therefore, the app does not need to pass any parameters in order to unregister for these activation kinds.
+
+```c++
+void UnregisterForActivation()
+{
+    // Get the AppLifecycle sub-object for activation registration.
+    ActivationRegistrationManager registrationManager = 
+        AppLifecycle::GetActivationRegistrationManager();
+
+    // Unregister one or more registered filetypes.
+    registrationManager.UnregisterForFileTypeActivation(
+        new string[]{ ".jpg", ".png", ".bmp" });
```

Comment:
C++ style would be to say `{ L".jpg", ...}` - `new string[]` is a C#-ism :)

Reply:
I'm pasting from my mocked C++ app, which builds ok...